### DISCUSSION TITLE
Typing reverse relations

### DISCUSSION THREAD

**Discussion by Winspear** [python/typing#1641] [Q&A]
Throughout my codebase I have a number of related_name relations defined. For most, django-stubs (latest release) seems to be generating the requisite annotations to make MyPy pass, but for certain models the annotation is not being generated. One example of this is a more complicated structure than a simple model to model, foreign key reference, as it uses an abstract base model, eg:
```
class BaseImage(models.Model):
    image = models.ImageField(blank=True)

class FruitImage(BaseImage):
    fruit = models.OneToOneField(
        Fruit,
        related_name="image")

class Fruit(models.Model):
    name = models.CharField(max_length=10)
```

With this particular structure, I am receiving the mypy error `"Fruit" has no attribute "image"`

Is there anything I need to do to properly annotate reverse relations of this form?

**Comment by sobolevn** [comment]
Please, report this to our tracker: https://github.com/typeddjango/django-stubs
